packages/scaffold-utils/src/SolanaTypesUtil.ts

```diff
@@ -0,0 +1,335 @@
+  signMessage: (message: Uint8Array) => Promise<Uint8Array> | Promise<{ signature: Uint8Array }>
+  signTransaction: (transaction: SolanaWeb3Transaction | VersionedTransaction) => Promise<{
+    signatures: { signature: Uint8Array }[]
+  }>
+  sendTransaction: (
+    transaction: SolanaWeb3Transaction | VersionedTransaction,
+    connection: Connection,
+    options?: SendTransactionOptions
+  ) => Promise<TransactionSignature>
+}
+
+export type Metadata = {
+  name: string
+  description: string
+  url: string
+  icons: string[]
+}
+
+export type CombinedProvider = W3mFrameProvider & Provider
+
+export type Chain = {
+  rpcUrl: string
+  explorerUrl: string
+  currency: string
+  name: string
+  chainId: string
+}
+
+// eslint-disable-next-line no-shadow
+export enum Tag {
+  Uninitialized = 0,
+  ActiveOffer = 1,
+  CancelledOffer = 2,
+  AcceptedOffer = 3,
+  FavouriteDomain = 4,
+  FixedPriceOffer = 5,
+  AcceptedFixedPriceOffer = 6,
+  CancelledFixedPriceOffer = 7
+}
+
+export interface Status {
+  Ok: null
+}
+
+export interface Meta {
+  err: null
+  fee: number
+  innerInstructions: TransactionInstruction[]
+  logMessages: string[]
+  postBalances: number[]
+  postTokenBalances: number[]
+  preBalances: number[]
+  preTokenBalances: number[]
+  rewards: null
+  status: Status
+}
+
+interface TransactionInstruction {
+  accounts: number[]
+  data: string
+  programIdIndex: number
+}
+
+export interface TransactionElement {
+  meta: Meta
+  transaction: TransactionTransaction
+}
+
+export interface TransactionTransaction {
+  message: Message
+  signatures: string[]
+}
+
+export interface Transaction {
+  message: Message
+  signatures: string[]
+}
+
+export interface TransactionResult {
+  meta: Meta
+  slot: number
+  transaction: Transaction
+}
+
+export interface Message {
+  accountKeys: string[]
+  header: Header
+  instructions: Instruction[]
+  recentBlockhash: string
+}
+
+export interface Header {
+  numReadonlySignedAccounts: number
+  numReadonlyUnsignedAccounts: number
+  numRequiredSignatures: number
+}
+
+export interface Instruction {
+  accounts: number[]
+  data: string
+  programIdIndex: number
+}
+
+export interface BlockResult {
+  blockHeight: number
+  blockTime: null
+  blockhash: string
+  parentSlot: number
+  previousBlockhash: string
+  transactions: TransactionElement[]
+}
+
+export interface AccountInfo {
+  data: string[]
+  executable: boolean
+  lamports: number
+  owner: string
+  rentEpoch: number
+}
+
+export type FilterObject =
+  | {
+      memcmp: {
+        offset: number
+        bytes: string
+        encoding?: string
+      }
+    }
+  | { dataSize: number }
+
+export interface TransactionInstructionRq {
+  programId: string
+  data: string
+  keys: {
+    isSigner: boolean
+    isWritable: boolean
+    pubkey: string
+  }[]
+}
+
+interface VersionedInstractionRq {
+  data: string
+  programIdIndex: number
+  accountKeyIndexes: number[]
+}
+
+export interface RequestMethods {
```

Comment:
Curious: do we manually create these types or copy over from somewhere? If we copy from somewhere, could you add a source to there within a JSDoc on top?